Completion/Unix/Command/_git

```diff
@@ -7130,12 +7130,27 @@ __git_deleted_files () {
 
 (( $+functions[__git_modified_files] )) ||
 __git_modified_files () {
-  __git_files --modified modified-files 'modified file' $*
+  __git_status_files
+}
+
+(( $+functions[__git_status_files] )) ||
+__git_status_files () {
+  local -a status_files=( ${"${(0)"$(git status -z)"}"} )
+
+  local -a unstaged_files_with_status=( ${(M)status_files:#?[^ ]*} )
+  local -a unstaged_files=( ${unstaged_files_with_status/?? /} )
+  _describe -t unstaged 'Unstaged' unstaged_files && ret=0
+
+  local -a staged_files_with_status=( ${(M)status_files:#[^ \?]*} )
+  local -a staged_files=( ${staged_files_with_status/?? /} )
+  _describe -t staged 'Staged' staged_files && ret=0
+
+  return $ret
 }
 
 (( $+functions[__git_other_files] )) ||
 __git_other_files () {
-  __git_files --others untracked-files 'untracked file' $*
+
```

Comment:
1. Same issue with `$*` as above.
2. Leaving the function empty would break `_git-clean`, which also calls this function.